framework/core/Project.pm

```diff
@@ -1064,6 +1087,7 @@ sub _ant_call {
     @_ >= 2 or die $ARG_ERROR;
     my ($self, $target, $option_str, $log_file) =  @_;
     $option_str = "" unless defined $option_str;
+    # TODO why are we checking for this if we never use it
```

Comment:
Line 1093 checks whether the build file exists. Do you mean that the code never uses the `_build_file` attribute?

Reply:
It seems like we check if `$self->{_build_file}` exists but then when running ant we use `$D4J_BUILD_FILE`.  I think we had a discussion about this previously but I would like to double check, does ant import the project build file (`$self-{_build_file}`) from the master (`$D4J_BUILD_FILE`)?